components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/beans/DataSourceDefinition.java

```diff
@@ -15,24 +15,41 @@
  */
 package org.wso2.carbon.datasource.core.beans;
 
-import org.w3c.dom.Element;
 import org.wso2.carbon.datasource.utils.DataSourceUtils;
+import org.wso2.carbon.kernel.annotations.Configuration;
+import org.wso2.carbon.kernel.annotations.Element;
+import org.yaml.snakeyaml.Yaml;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
 
-import javax.xml.bind.annotation.XmlAnyElement;
-import javax.xml.bind.annotation.XmlAttribute;
-import javax.xml.bind.annotation.XmlRootElement;
 
 /**
  * Class holding the data source definition.
  */
-@XmlRootElement(name = "definition")
+@Configuration(description = "data source definition")
 public class DataSourceDefinition {
 
-    private String type;
+    @Element(description = "data source type", required = true)
+    private String type = "RDBMS";
+
+    public DataSourceDefinition() {
+        Map<String, String> configMap = new LinkedHashMap<String, String>();
+        configMap.put("jdbcUrl", "jdbc:h2:./database/WSO2_CARBON_DB;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000");
```

Comment:
This default values are for the configure documentation. we don't use these keys anyway else. only in this class. do we need to specify keys as constants.